### DISCUSSION TITLE
CKAN Upgrade | 2.7.2 to 2.10.1

### DISCUSSION THREAD

**Discussion by pachori91** [ckan/ckan#8247] [Q&A]
Hi Team,

I have a Docker based ckan2.7.2 running in production environment and want to upgrade to version 2.10.4.
i just checked in CKAN documentation below are the possible upgradation-
[Upgrading a CKAN 2 package install to a new patch release](https://docs.ckan.org/en/2.10/maintaining/upgrading/upgrade-package-to-patch-release.html)
[Upgrading a CKAN 2 package install to a new minor release](https://docs.ckan.org/en/2.10/maintaining/upgrading/upgrade-package-to-minor-release.html)
[Upgrading a source install](https://docs.ckan.org/en/2.10/maintaining/upgrading/upgrade-source.html)

But unable to find upgradation steps in case of docker based setup.

Can you guys please provide me any suggestion for same? How i can proceed further.

Thanks

**Comment by pachori91** [comment]
anyone please reply

**Comment by muhammetcetinyamac** [comment]
Hello, 

There are major updates available in the CKAN version you are planning to migrate to. As seen in the documentation, certain structures such as IRoutes have been deprecated after CKAN 2.9.5. These are now continued with blueprints. You are aiming to transition to CKAN 2.10.x. If you are using any plugins, they might pose significant issues as well. For example, the datarequest plugin does not function after CKAN 2.9.x. You may encounter serious problems with many such plugins. Similarly, there are major changes on the database side. Such an update will require a very delicate operation. My advice is to read CKAN's changelog and review your database, plugins, and other structures accordingly.

Ckan v.2.10.4 2024-03-13 Changelog: https://docs.ckan.org/en/2.10/changelog.html

Best regards
Muhammed Çetinyamaç
